packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -132,8 +136,27 @@ def is_task_running(self, task_name: TaskName) -> bool:
         managed_tasks_ids = list(self._tasks_groups[task_name].keys())
         return len(managed_tasks_ids) > 0
 
+    def list_tasks(self, with_task_context: Optional[TaskContext]) -> list[TrackedTask]:
+        tasks = []
+        for task_group in self._tasks_groups.values():
+            if not with_task_context:
+                tasks.extend(task_group.values())
+            else:
+                tasks.extend(
+                    [
+                        task
+                        for task in task_group.values()
+                        if task.task_context == with_task_context
```

Comment:
so this list that tasks that "fit the context"?? 

THOUGHTS: 
1. since you use this to filter, should not be the ``task_context`` frozen once created? Analogous to the app/request storage ... ? SEE [frozendict](https://github.com/Marco-Sulla/python-frozendict) or [How to make an immutable](https://adamj.eu/tech/2022/01/05/how-to-make-immutable-dict-in-python/) 
1. Don't you want to be more explicit? e.g. you can define a "minimum required context" (e.g. one that contains user_id) and then use that to filter. This way you still filter and at the same time give the library caller the flexibility to extend the task context with other info that is not used to filter